sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -916,6 +919,49 @@ public Symbol visitObjectLiteral(ObjectLiteral node, ExpressionAnalysisContext c
                 context);
         }
 
+        @Override
+        public Symbol visitIntervalLiteral(IntervalLiteral node, ExpressionAnalysisContext context) {
+            String value = node.getValue();
+            if (value == null || value.isEmpty() || value.isBlank()) {
+                throw new IllegalArgumentException("Invalid value " + value);
+            }
+            int time = Integer.parseInt(value);
+
+            int months = 0;
+            int days = 0;
+            double seconds = 0;
+
+            //TODO include getEndField TO clause
+            switch (node.getStartField()) {
+                case YEAR:
+                case MONTH:
+                    months = Math.toIntExact(Period.ofMonths(time).toTotalMonths());
+                    break;
+                case DAY:
+                    days = Math.toIntExact(Duration.ofDays(time).toDays());
+                    break;
+                case HOUR:
+                    seconds = Math.toIntExact(Duration.ofHours(time).getSeconds());
+                    break;
+                case MINUTE:
+                    seconds = Math.toIntExact(Duration.ofMinutes(time).getSeconds());
+                    break;
+                case SECOND:
+                    seconds = Duration.ofSeconds(time).getSeconds();
+                    break;
+                default:
+                    throw new IllegalArgumentException("Invalid start field");
+            }
+
+            MonthDaySecondInterval MonthDaySecondInterval = new MonthDaySecondInterval(seconds, days, months);
+
+            if (node.getSign() == IntervalLiteral.Sign.NEGATIVE) {
```

Comment:
Instead of doing the negation after the initial `new MonthDaySecondInterval(seconds, days, months)` there could be one `if/else` block that does the `new Month..` initialization. So that we don't have 1 useless positive initialization if a negative sign is used.